homeassistant/components/alexa/smart_home.py

```diff
@@ -1236,16 +1251,7 @@ def extract_entity(funct):
         ATTR_ENTITY_ID: entity.entity_id
     }, blocking=False, context=context)
 
-    # Alexa expects a lockState in the response, we don't know the actual
-    # lockState at this point but assume it is locked. It is reported
-    # correctly later when ReportState is called. The alt. to this approach
-    # is to implement DeferredResponse
-    properties = [{
-        'name': 'lockState',
-        'namespace': 'Alexa.LockController',
-        'value': 'LOCKED'
-    }]
-    return api_message(request, context={'properties': properties})
```

Comment:
So if we remove this, the property will get added later but will it say "locked"? I didn't test locks specifically, but it seemed with thermostats (which have a similar problem) that although the relevant services had been called to change the temperature, that isn't yet reflected in the entity state by the time the properties are added to the response.

I'm a bit fuzzy on the home assistant internals, but I guess it's because services are executed asynchronously. Is that right? In #17965 I just added a method for handlers to mark the properties they've changed, which then takes precedence over any automatically set properties.

From a UX perspective, I think not waiting for the state change to happen is the right way to go. Those few ms shaved off the response time make a big difference. IIRC, there is a `DeferredResponse` we could implement which might have a different behavior, but I'm pretty happy with things as they are.

Reply:
yeah I was unsure about this. So faking state changes feel like a hack to me. I rather implement deferred change. We can make it do the real change if we pass `blocking=True` to the service call code. I feel that would be the _right_ thing to do, and if we can't make it in time, we can return a deferred state.